lib/aytests/image_builder.rb

```diff
@@ -92,7 +93,7 @@ def upgrade(autoinst, iso_url)
       setup_definition(:upgrade)
       change_boot_order
       backup_image
-      build
+      build(build_environment(autoinst_path.sub_ext(".linuxrc")))
```

Comment:
Is there any situation in which you want to call `build` with a different `build_environment`?

I mean, why not just leave `build` without parameters and call `build_environment` inside?

Same applies to `autoinst_path.sub_ext(".linuxrc")`. Do you ever use a different file? If not, I would move that part to the `build_environment` method.